updating of calc/meas/proc/.count/intensity definitions

rowlesmr opened the pull request:
Was initially looking just to address #42 .

But I now have other questions:
- `pd_meas.counts/intensity_` should be `Recorded`, not `Derived` - Addressed in this initial commit.
- Some ambiguity in definitions of `_pd_proc.intensity_norm` and `_pd_proc.intensity_net`: which one contains a bkg? From the definition of `_pd_calc.intensity_net`, I'm guessing `_pd_proc.intensity_net` is background-corrected and normalisation-corrected, and the other is just normalisation-corrected.
- Should all `_pd_proc.intensity_*` and `_pd_calc.intensity_*` have their values limited to be > 0 (as is currently)? ( c.f. discussion in [cif_core](https://github.com/COMCIFS/cif_core/pull/315#issuecomment-1328622630))
- The definitions of `_pd_meas.counts/intensity_background/container/monitor` say (amongst other things...)
   -  >    Corrections for background, detector dead time etc. should not have been made to these values. Instead use `_pd_proc.intensity_*` for corrected diffractograms.
   - There is no direct `_pd_proc.` equivalent to these data names. Does this above paragraph mean that you should use `_pd_proc.intensity_net/norm/total` as appropriate, after doing the corrections? or are there missing `_pd_proc.intensity_background/container/monitor` data names?

jamesrhester commented:
I like the changes made so far in this pull request.

`_pd_calc.intensity_net` and `_pd_proc.intensity_net` should certainly be calculated in the same way in order to be directly comparable, so I agree that they should be both normalised and background-subtracted.

I think restricting `_pd_calc.intensity_net` values to be >= zero is defensible. I would leave the other one unrestricted in recognition of the fact that current software does not always treat values close to zero in a way that would avoid small negative values, which would involve a bit of not-very-advanced Bayesian footwork. I am open to be convinced otherwise and don't really care either way.

The `_pd_proc` equivalents are missing because, as suggested, these `_pd_meas` data names have been used to calculate the total/net/norm intensities which the refinement works on. I don't see any need to add the equivalents.

If everyone is happy with these replies I think we can merge these changes.

rowlesmr commented:
I've removed the enumeration range on `_pd_proc.intensity_net`, added some words in the description of `_pd_proc.intensity_background/container/monitor` talking about `_pd_proc.intensity_net/norm/total`, and clarified the bkg/norm correction for `_pd_proc.intensity_norm/net`.

I think that covered everything.

jamesrhester reviewed (approved):
Happy with these changes.
